state-chain/runtime/src/chainflip.rs

```diff
@@ -315,21 +315,29 @@ impl TransactionBuilder<Bitcoin, BitcoinApi<BtcEnvironment>> for BtcTransactionB
 pub struct SolanaTransactionBuilder;
 impl TransactionBuilder<Solana, SolanaApi<SolEnvironment>> for SolanaTransactionBuilder {
 	fn build_transaction(
-		_signed_call: &SolanaApi<SolEnvironment>,
+		signed_call: &SolanaApi<SolEnvironment>,
 	) -> <Solana as Chain>::Transaction {
-		todo!()
+		signed_call.transaction.clone()
 	}
+
 	fn refresh_unsigned_data(_tx: &mut <Solana as Chain>::Transaction) {
-		todo!()
+		// Transaction fee on SOlana is a fixed 5k lamports per signature. Since we always have
+		// a single signature, therefore the transaction fee does not change.
 	}
+
 	fn calculate_gas_limit(_call: &SolanaApi<SolEnvironment>) -> Option<U256> {
-		todo!()
+		// Solana sets computation limits via the `compute_limit` attribute, instead of
+		// via transaction fee. Therefore this mechanism is not used for Solana.
```

Comment:
This is true for regular swaps but not for CCM Swaps. For the later, it will be similar to Ethereum/Arbitrum where we should calculate the gas_limit from the output of the gas swap and set that as the gas limit (`compute_limit`).
However, that also requires changes in the Transaction Builder. I'd first leave this in a state that it works for regular swaps and I'll open a PR where we tackle the gas limit for CCM swaps.

Reply:
Reason being that the priority is getting regular swaps tested and we can implement CCM-specific logic separately.

Reply:
Ok sounds good!

Reply:
Added it here: https://linear.app/chainflip/issue/PRO-1479/solana-ccm-gas-limit-calculation
Please add some reference to that linear ticket in the code.